tests/FixtureBuilder/ExpressionLanguage/Lexer/ReferenceEscaperLexerTest.php

```diff
@@ -66,5 +66,7 @@ public function provideValues()
         yield 'string with a reference with members' => ['bar @foo baz'];
 
         yield 'reference in a middle of a word' => ['email@example', 'email\\@example'];
+
+        yield 'email ending with a digit' => ['email1@example', 'email1\\@example'];
```

Comment:
given the issues we had, feels like we should add a few more tests with exotic emails (with non ASCII characters for example)